pinecone/data/features/bulk_import.py

```diff
@@ -0,0 +1,193 @@
+from enum import Enum
+from typing import Optional, Literal, Iterator, List, Type, cast
+
+from pinecone.config.config import ConfigBuilder
+from pinecone.core_ea.openapi.db_data import ApiClient
+from pinecone.core_ea.openapi.db_data.api.bulk_operations_api import BulkOperationsApi
+from pinecone.core_ea.openapi.shared import API_VERSION
+
+from pinecone.utils import parse_non_empty_args, install_json_repr_override, setup_openapi_client
+
+from pinecone.core_ea.openapi.db_data.models import (
+    StartImportRequest,
+    StartImportResponse,
+    ImportListResponse,
+    ImportModel,
+    ImportErrorMode as ImportErrorModeClass,
+)
+
+for m in [StartImportResponse, ImportListResponse, ImportModel]:
+    install_json_repr_override(m)
+
+ImportErrorMode: Type[Enum] = cast(
+    Type[Enum], Enum("ImportErrorMode", ImportErrorModeClass.allowed_values[("on_error",)])
+)
+
+
+class ImportFeatureMixin:
+    def __init__(self, **kwargs):
+        config = ConfigBuilder.build(
+            **kwargs,
+        )
+        openapi_config = ConfigBuilder.build_openapi_config(config, kwargs.get("openapi_config", None))
+
+        if kwargs.get("__import_operations_api", None):
+            self.__import_operations_api = kwargs.get("__import_operations_api")
+        else:
+            self.__import_operations_api = setup_openapi_client(
+                api_client_klass=ApiClient,
+                api_klass=BulkOperationsApi,
+                config=config,
+                openapi_config=openapi_config,
+                pool_threads=kwargs.get("pool_threads", 1),
+                api_version=API_VERSION,
+            )
+
+    def start_import(
+        self,
+        uri: str,
+        integration_id: Optional[str] = None,
+        error_mode: Optional[Literal["CONTINUE", "ABORT"]] = "CONTINUE",
+    ) -> StartImportResponse:
+        """Import data from a storage provider into an index. The uri must start with the scheme of a supported
+        storage provider. For buckets that are not publicly readable, you will also need to separately configure
```

Comment:
Maybe so users have a better understanding of what we mean when we say "storage provider," you write "e.g. "S3"" or something?

Reply:
Currently only S3 is supported, but I don't want to write that explicitly because I think it will quickly go out of date. The error message the API sends back is relatively helpful if you get this wrong.